presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaMetadata.java

```diff
@@ -194,7 +200,32 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns)
     {
         KafkaTableHandle handle = convertTableHandle(table);
-        ConnectorTableLayout layout = new ConnectorTableLayout(new KafkaTableLayoutHandle(handle));
+
+        Long startTs = null;
+        Long endTs = null;
+        Optional<Map<ColumnHandle, Domain>> domains = constraint.getSummary().getDomains();
+        log.info("TableLayout");
+        if (domains.isPresent()) {
+            Map<ColumnHandle, Domain> columnHandleDomainMap = domains.get();
+            for (Map.Entry<ColumnHandle, Domain> entry : columnHandleDomainMap.entrySet()) {
+                if (entry.getKey() instanceof KafkaColumnHandle && ((KafkaColumnHandle) entry.getKey()).getName().equals(KafkaInternalFieldDescription.OFFSET_TIMESTAMP_FIELD.getName())) {
+                    Range span = entry.getValue().getValues().getRanges().getSpan();
+                    Marker low = span.getLow();
+                    Marker high = span.getHigh();
+                    if (!low.isLowerUnbounded()) {
+                        startTs = (Long) low.getValue();
+                    }
+                    if (!high.isUpperUnbounded()) {
+                        endTs = (Long) high.getValue();
+                    }
+                }
+
+                log.info("K: %s\tV: %s", entry.getKey().toString(), entry.getValue().toString());
+            }
+        }
+
+        log.info("startTs: %s, endTs: %s", startTs, endTs);
```

Comment:
null cases are handled at `KafkaTableLayoutHandle`. 
Are you saying no queries should go beyond without providing a start and end offset timestamp?